signer/src/eth.rs

```diff
@@ -6,11 +6,11 @@
 
 use crate::ecdsa;
 use alloc::format;
-use alloc::string::String;
 use core::fmt::{Display, Formatter};
 use core::str::FromStr;
 use keccak_hash::keccak;
 use secp256k1::Message;
+use subxt_core::utils::AccountId20;
```

Comment:
Let's avoid importing `subxt-core` here, and instead copy the pattern from eg `sr25519.rs` where we have a standalong `PublicKey` struct and then in the `subxt_compat` section we pull in `subxt-core` and have the relevant `From` impls etc to make `PublicKey` play nice with `subxt`'s `AccountId20` :)

Reply:
what about the tests for eth? they will be enable only if "subxt" feature is also enabled

Reply:
Ah the tests are no problem; currently we just assume the default features are enabled for them, which allows subxt-core anyways. I just think that we can avoid needing subxt-core by default for "eth" signing to fall in line with the other algorithms :)